src/WebAPI/Controllers/FilmsController.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using WebAPI.Models;
+
+namespace WebAPI.Controllers
+{
+    [Produces("application/json")]
+    [Route("api/[controller]")]
+    public class FilmsController : Controller
+    {
+        // GET /films/now-playing
+        [HttpGet]
+        [Route("now-playing")]
+        public IEnumerable<FilmModelResponse> GetNowPlayingFilms()
+        {
+            List<FilmModelResponse> nowPlayingFilms = new List<FilmModelResponse>();
+            return nowPlayingFilms;
+        }
+
+        // GET /films
+        [HttpGet]
+        [Route("")]
+        public IEnumerable<FilmModelResponse> Get()
+        {
+            List<FilmModelResponse> Films = new List<FilmModelResponse>();
+            return Films;
+        }
+
+        // GET /films/{name}
+        [HttpGet("{name:string}")]
+        public IEnumerable<FilmModelResponse> Get([FromQuery]string name)
+        {
+            List<FilmModelResponse> FilmsByName = new List<FilmModelResponse>();
+            return FilmsByName;
+        }
+
+        // GET /films/{id}
+        [HttpGet("{id:int}")]
+        public FilmModelResponse Get(int id)
+        {
+            FilmModelResponse film = new FilmModelResponse();
+            return film;
+        }
+
+        // GET /films/{id}/sessions
+        [HttpGet("{id:int}/sessions")]
+        public IEnumerable<SessionModelResponseForFilmAPI> GetSessions(int id)
+        {
+            List<SessionModelResponseForFilmAPI> sessions = new List<SessionModelResponseForFilmAPI>();
+            return sessions;
+        }
+
+        // POST /films/search-films
```

Comment:
From my point of view this one includes public IEnumerable<FilmModelResponse> Get([FromQuery]string name)